airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/substream_slicer.py

```diff
@@ -54,22 +54,38 @@ def update_cursor(self, stream_slice: StreamSlice, last_record: Optional[Record]
                 cursor.update({parent_stream_config.stream_slice_field: slice_value})
         self._cursor = cursor
 
-    def request_params(self) -> Mapping[str, Any]:
+    def request_params(
+        self,
+        stream_state: Optional[StreamState] = None,
+        stream_slice: Optional[StreamSlice] = None,
+        next_page_token: Optional[Mapping[str, Any]] = None,
+    ) -> Mapping[str, Any]:
         return self._get_request_option(RequestOptionType.request_parameter)
 
-    def request_headers(self) -> Mapping[str, Any]:
+    def request_headers(
+        self,
+        stream_state: Optional[StreamState] = None,
```

Comment:
very small nit: is it more pythonic to use `**kwargs` insteads of naming unused keyword args? seems more maintainable that way but I'm not sure if I'm missing something

Reply:
@sherifnada 
the problem with using `**kwargs` in the interface is that the user of the class cannot know what parameters are without knowing the concrete implementation - https://en.wikipedia.org/wiki/Law_of_Demeter.

That being said, the stream slicer, the paginator, and the requester are never used interchangeably (and can't because they expect different parameters) so I think it makes more sense for them not to inherit from `RequestOptionsProvider`.
is that reasonable?

Reply:
ah I missed that this was an ABC -- this makes sense. I thought it was an implementing class.